packages/datadog-plugin-jest/src/jest-jasmine2.js

```diff
@@ -0,0 +1,199 @@
+const { promisify } = require('util')
+
+const id = require('../../dd-trace/src/id')
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+const { SAMPLING_PRIORITY, SPAN_TYPE, RESOURCE_NAME } = require('../../../ext/tags')
+const { AUTO_KEEP } = require('../../../ext/priority')
+const {
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS,
+  CI_APP_ORIGIN,
+  getTestEnvironmentMetadata,
+  finishAllTraceSpans
+} = require('../../dd-trace/src/plugins/util/test')
+
+function getTestSpanTags (tracer, testEnvironmentMetadata) {
+  const childOf = tracer.extract('text_map', {
+    'x-datadog-trace-id': id().toString(10),
+    'x-datadog-parent-id': '0000000000000000',
+    'x-datadog-sampled': 1
+  })
+
+  const commonSpanTags = {
+    [TEST_TYPE]: 'test',
+    [SAMPLING_RULE_DECISION]: 1,
+    [SAMPLING_PRIORITY]: AUTO_KEEP,
+    [SPAN_TYPE]: 'test',
+    ...testEnvironmentMetadata
+  }
+  return {
+    childOf,
+    commonSpanTags
+  }
+}
+
+function createWrapIt (tracer, globalConfig, globalInput, testEnvironmentMetadata) {
+  return function wrapIt (it) {
+    return function itWithTrace (description, specFunction, timeout) {
+      let oldSpecFunction = specFunction
+      if (specFunction.length) {
+        oldSpecFunction = promisify(oldSpecFunction)
+      }
+
+      const { childOf, commonSpanTags } = getTestSpanTags(tracer, testEnvironmentMetadata)
+
+      const testSuite = globalInput.jasmine.testPath.replace(`${globalConfig.rootDir}/`, '')
+
+      const newSpecFunction = tracer.wrap(
+        'jest.test',
+        {
+          type: 'test',
+          childOf,
+          tags: { ...commonSpanTags, [TEST_SUITE]: testSuite }
+        },
+        async (done) => {
```

Comment:
A lot of this function is repeated in the 2 files. Can some of it be extracted to some common util?

Reply:
the little differences make it quite complicated. I've extracted a function to set the suppressed errors, but I don't think the rest can be extracted